src/modules/auth/auth.controller.ts

```diff
@@ -31,4 +42,21 @@ export class AuthController {
   login(@Body() loginDto: LoginDto) {
     return this.authService.login(loginDto);
   }
+
+  @Get('google/redirect')
+  @UseGuards(AuthGuard('google'))
+  @ApiOperation({
+    summary: 'Google Authentication Callback',
+    description: 'Callback for Google OAuth2 login',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'User info from Google',
+  })
+  async googleAuthRedirect(@GetUser() user: User) {
+    if (!user) {
+      throw new NotFoundException('User not found');
```

Comment:
We are not searching for user here, just checking if request has valid credentials. Therefore, it would be better to throw ```UnauthorizedException```.